### PR TITLE
Add API Documents

### PR DESCRIPTION
**Pull request by chaosi-zju**
**What type of PR is this?**

/kind documentation

**What this PR does / why we need it**:

add api documents

**Which issue(s) this PR fixes**:

part of https://github.com/karmada-io/karmada/issues/4090

**Special notes for your reviewer**:

@XiShanYongYe-Chang @RainbowMango 

[Quick Preview](https://deploy-preview-476--karmada.netlify.app/docs/next/reference/api-documents/cluster-resources/cluster-v1alpha1)

If you want to generate api document, you can do like this:

1) update [`api/v1.7/swagger.json`] and [`config/v1.7/toc.yaml`] in [reference-docs](https://github.com/chaosi-zju/reference-docs). (temporary plan, to be adjusted after holidays)

~2. clone this latest version repo, run `bash reference-api.sh` in a linux machine.~ (I cancelled submitting this script)

2) run following commands to generate the doc：
```bash
#!/bin/bash

git clone https://github.com/chaosi-zju/reference-docs.git
cd reference-docs || exit 1

# generate api document
make

# delete all the carriage Return (CR)
sed -i 's/\r//g' `find kwebsite -name "*.md"`
# remove the redundant linefeed (LF)
sed -i '/^$/N;/^\n$/D' `find kwebsite -name "*.md"`

# move the generated docs to Docusaurus doc path
rm -rf ../docs/reference/api-documents
mv -f kwebsite/content/en/docs ../docs/reference/api-documents
cd .. && rm -rf reference-docs
```
3) git add/commit/push

### CODE REVIEW

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Thanks a lot!
In terms of content and style presented, LGTM.
There may be some points that can be optimized in the future. Can you help explain them in the PR description? So that we can complete it one by one in the subsequent work plan.

Ask @Affan-7 and @RainbowMango to help take a look.

**Comment by chaosi-zju** [comment]
> There may be some points that can be optimized in the future. Can you help explain them in the PR description? 

No, all the optimization points we have discussed has already been implemented~

File: reference-api.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (chaosi-zju) [reply reference-api.sh:1]: > Can we submit it in another PR?

ok

> I understand that this script only completes part of the API document generation work

Actually, this script done all API document generation work, just one line `bash reference-api.sh` can finish the doc generation, however this script depends on [reference-docs](https://github.com/chaosi-zju/reference-docs).

**Comment by Affan-7** [comment]
The output looks good to me.

File: sidebars.js [hunk 2]
```diff
@@ -290,6 +290,92 @@ module.exports = {
                         "reference/instrumentation/metrics",
                     ],
                 },
+                {
+                    type: "category",
+                    label: "API Document",
```
Comment (RainbowMango) [inline sidebars.js:295]:
```suggestion
                    label: "Karmada API",
```

File: sidebars.js [hunk 3]
```diff
@@ -290,6 +290,92 @@ module.exports = {
...
+                    items: [
+                        {
+                            type: "category",
+                            label: "Auto Scaling Resources",
+                            items: [
+                                "reference/api-documents/auto-scaling-resources/cron-federated-hpa-v1alpha1",
+                                "reference/api-documents/auto-scaling-resources/federated-hpa-v1alpha1",
```
Comment (RainbowMango) [inline sidebars.js:302]:
```suggestion
                                "reference/karmada-api/auto-scaling-resources/cron-federated-hpa-v1alpha1",
                                "reference/karmada-api/auto-scaling-resources/federated-hpa-v1alpha1",
```
same as below.

**Review by RainbowMango — COMMENTED** [review]
Good job, just some suggestions.

File: sidebars.js (hunk 3 above)
Comment (chaosi-zju) [reply sidebars.js]: done, [click](https://deploy-preview-476--karmada.netlify.app/docs/next/reference/karmada-api/auto-scaling-resources/cron-federated-hpa-v1alpha1)

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**